addOns/exim/src/main/java/org/zaproxy/addon/exim/log/LogsImporter.java

```diff
@@ -226,20 +247,38 @@ private static void processZapLogs(List<String> parsedRequestAndResponse)
                                 tempResponseHeader,
                                 tempResponseBody);
                 httpMessage.setResponseFromTargetHost(true);
-                createHistoryReferenceAndAddToTree(httpMessage);
+                createHistoryReferenceAndAddToTree(httpMessage, LogType.ZAP);
             }
         }
     }
 
-    private static void createHistoryReferenceAndAddToTree(HttpMessage message) {
+    private static void createHistoryReferenceAndAddToTree(HttpMessage message, LogType logType) {
         Session currentSession = Model.getSingleton().getSession();
 
         try {
             HistoryReference historyRef =
                     new HistoryReference(currentSession, HistoryReference.TYPE_ZAP_USER, message);
             addToTree(historyRef);
+            if (LogType.ZAP.equals(logType)) {
+                Stats.incCounter(ExtensionExim.STATS_PREFIX + STATS_ZAP_FILE_MSG);
+            } else {
+                Stats.incCounter(ExtensionExim.STATS_PREFIX + STATS_MODSEC2_FILE_MSG);
+            }
         } catch (DatabaseException | HttpMalformedHeaderException | NullPointerException e) {
             LOG.error(e.getMessage(), e);
+            if (LogType.ZAP.equals(logType)) {
+                Stats.incCounter(ExtensionExim.STATS_PREFIX + STATS_ZAP_FILE_MSG_ERROR);
+            } else {
+                Stats.incCounter(ExtensionExim.STATS_PREFIX + STATS_MODSEC2_FILE_MSG_ERROR);
+            }
+        }
+    }
+
+    private static void updateOutput(String messageKey, String filePath) {
+        if (View.isInitialised()) {
+            StringBuilder sb = new StringBuilder();
```

Comment:
Could initialise to some higher value than the default, the file path will surely be bigger than that. (Same comment for the other import.)